packages/myst-config/src/errorRules/validators.ts

```diff
@@ -0,0 +1,63 @@
+import type { ValidationOptions } from 'simple-validators';
+import {
+  defined,
+  incrementOptions,
+  validateChoice,
+  validateList,
+  validateObjectKeys,
+  validateString,
+} from 'simple-validators';
+import type { ErrorRule } from './types.js';
+
+const ERROR_RULE_KEY_OBJECT = {
+  required: ['id'],
+  optional: ['severity'],
+  alias: {
+    rule: 'id',
+  },
+};
+
+export function validateErrorRule(input: any, opts: ValidationOptions): ErrorRule | undefined {
+  if (typeof input === 'string') {
+    input = { id: input };
+  }
+  const value = validateObjectKeys(input, ERROR_RULE_KEY_OBJECT, {
+    ...opts,
+    suppressWarnings: true,
+    keepExtraKeys: true,
+  });
+  if (value === undefined) return undefined;
+  const id = validateString(value.id, incrementOptions('id', opts));
+  if (!id) return undefined;
+  const output: ErrorRule = { id };
+  if (defined(value.severity)) {
+    const choice = validateChoice(value.severity, {
+      ...incrementOptions('severity', opts),
+      choices: ['ignore', 'warn', 'error'],
+    }) as ErrorRule['severity'];
+    if (choice) output.severity = choice;
+  }
+  // Put the unknown fields back on the object
```

Comment:
My understanding is these unknown fields are intended to differentiate between multiple errors of the same `ruleId`? This feels maybe too flexible? To filter on these different entries, we are going to need to expand the line in `addWarningForFile`:
```typescript
const handler = config?.error_rules?.find((rule) => rule.id === opts.ruleId);
```
to something more like
```typescript
const handler = config?.error_rules?.find((rule) => {
  return rule.id === opts.ruleId && someOtherComparison(rule, opts);
});
```
If we allow arbitrary data, that `someOtherComparison` function becomes complicated and potentially rule-id-dependent. If instead we just have, say, `key`, then it's just an equality check. (Of course, we will still need rule-id-dependent key generation, but that can happen where the rule is actually getting violated, and by the time we get here, it's just a string.)

That also makes user feedback in the logging easier: "To suppress errors of this type, add `my-rule-id` to `error_rules`; to suppress this error only, add `key: my-rule-key`"

Reply:
Being a little stricter on this would allow:

```yaml
error_rules:
  - rule: doi-check
    severity: ignore
    keys:
      - bibKey1
      - bibKey2
      - bibKey3
```

to coerce to:

```yaml
error_rules:
  - rule: doi-check
    severity: ignore
    key: bibKey1
  - rule: doi-check
    severity: ignore
    key: bibKey2
  - rule: doi-check
    severity: ignore
    key: bibKey3
```

Which I think is nice? Easy for the user, easy for the handling logic.